### PR TITLE
fix sphinx errors

### PR DESCRIPTION
**Pull request by styamamoto**
Fixed sphinx errors but this work is WIP.

Some warnings are still remain because;
- `__init__.py` is not defined correctly.  [#182](https://github.com/aistairc/aiaccel/issues/182)
- Equations in `developer_guide/custom_optimizer.md`
 is not rendered correctly as html.
- I couldn't remove `WARNING: Definition list ends without a blank line; unexpected unindent.` arise from docstring of `aiaccel.config.BaseConfig.get_property`.

### CODE REVIEW

**Comment by styamamoto** [comment]
All of warnings and errors on docs-building are removed.
Note, some of extention and configurations are added to docs/source/conf.py.

File: aiaccel/config.py [hunk 1]
```diff
@@ -18,10 +18,9 @@
 
 
 class BaseConfig(metaclass=ABCMeta):
-    """
-    Fork by confile: https://github.com/777nancy/confile
+    """An interface for all config classes.
 
-    This is an interface for all config classes.
+    Fork by confile: https://github.com/777nancy/confile
```
Comment (yoshipon) [inline aiaccel/config.py:23]: これは残しておいた方が良いかと．多分まだconfileだった自体の名残があるってことですよね？ あれ違う…？

**Review by yoshipon — CHANGES_REQUESTED** [review]
気になっていた不備がキレイになってて助かります．

File: aiaccel/config.py (hunk 1 above)
Comment (styamamoto) [reply aiaccel/config.py:23]: 比較のページだとちょっとわかりづらいですが，文章自体は残っています．
(文の順番は入れ替わっていますが)

https://github.com/aistairc/aiaccel/blob/0dbbc6df0daae03988c2c7cc7ae33a70c6340835/aiaccel/config.py#L20-L24

**Review by yoshipon — APPROVED** [review]

**Pull request merged**